components-api/src/test/scala/pl/touk/nussknacker/engine/api/typed/TypingResultSpec.scala

```diff
@@ -147,6 +148,15 @@ class TypingResultSpec extends FunSuite with Matchers with OptionValues with Ins
     commonSuperTypeFinder.commonSupertype(Typed(Typed[Dog], Typed[Cactus]), Typed[Cat]) shouldEqual Typed[Pet]
   }
 
+  test("common supertype with generics") {
+    commonSuperTypeFinder.commonSupertype(Typed[Comparable[Number]], Typed[Integer]) shouldEqual Typed[Comparable[Number]]
+    commonSuperTypeFinder.commonSupertype(Typed[Integer], Typed[Comparable[Number]]) shouldEqual Typed[Comparable[Number]]
+    commonSuperTypeFinder.commonSupertype(Typed[util.List[Integer]], Typed[util.List[Number]]) shouldEqual Typed[util.List[Number]]
+    commonSuperTypeFinder.commonSupertype(Typed[util.List[Integer]], Typed[util.Collection[Number]]) shouldEqual Typed[util.Collection[Number]]
+    // below weird examples which will work
+    commonSuperTypeFinder.commonSupertype(Typed[util.List[Number]], Typed[util.Collection[Integer]]) shouldEqual Typed[util.Collection[Number]]
```

Comment:
this does not look that weird ;)
`trait A[T]`
`trait B[U] extends A[String]`
` commonSuperType(Typed[B[Int]], Typed[A[String]]) == Typed[A[Unknown]]`

Reply:
I've added this test now